### PR TITLE
Feature/babylon dealing bids

### PR DESCRIPTION
**Pull request by timlaengert**

### CODE REVIEW

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs [hunk 1]
```diff
@@ -154,21 +174,23 @@ public override async Task JoinGame(JoinGameRequest request, IServerStreamWriter
                 n.Status.Success = false;
                 n.Status.ErrorNum = (int)ErrorCode.GAME_NOT_FOUND;
                 n.Status.ErrorText = "No Game Exists";
-                await responseStream.WriteAsync(n);
+                // await responseStream.WriteAsync(n);
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:177]: remove comment if not needed

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs [hunk 2]
```diff
@@ -154,21 +174,23 @@ public override async Task JoinGame(JoinGameRequest request, IServerStreamWriter
...
-            RpcClient client = new RpcClient(responseStream, request.Name);
             game.AddClient(client);
             //game.Clients.Add(client);
 
             // Send A Join Game Response
             JoinGameResponse jgr = new JoinGameResponse();
             jgr.Token = client.Token;
             n.JoinResponse = jgr;
-            await responseStream.WriteAsync(n);
+            // await responseStream.WriteAsync(n);
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:191]: remove comment if not needed

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs [hunk 3]
```diff
@@ -209,7 +231,8 @@ public override async Task JoinGame(JoinGameRequest request, IServerStreamWriter
                 if (c is RpcClient)
                 {
                     RpcClient rpcClient = (RpcClient)c;
-                    tasks.Add(rpcClient.Stream.WriteAsync(notification));
+                    // tasks.Add(rpcClient.Stream.WriteAsync(notification));
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:234]: remove comment if not needed

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs [hunk 4]
```diff
@@ -345,14 +376,27 @@ public override async Task JoinGame(JoinGameRequest request, IServerStreamWriter
 
                     Notification n = new Notification();
                     n.Hand = hand;
-                    tasks.Add(rpcClient.Stream.WriteAsync(n));
+                    // tasks.Add(rpcClient.Stream.WriteAsync(n));
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:379]: remove comment if not needed

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs [hunk 5]
```diff
@@ -345,14 +376,27 @@ public override async Task JoinGame(JoinGameRequest request, IServerStreamWriter
...
             Log.Debug("Deal Notification complete for game " + game.Name);
         }
 
+        public async Task RequestBid(Game.Game game, Client bidder) {
+            if (bidder is RpcClient) {
+                RpcClient rpcClient = (RpcClient)bidder;
+
+                Notification n = new Notification();
+                n.BidRequest = new BidRequest();
+                // await rpcClient.Stream.WriteAsync(n);
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:394]: remove comment if not needed

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs [hunk 6]
```diff
@@ -345,14 +376,27 @@ public override async Task JoinGame(JoinGameRequest request, IServerStreamWriter
...
+                await SendNotification(rpcClient, n);
+                Log.Debug("Request Bid Notification complete for game " + game.Name);
+            }
+        }
+
         public async Task SendCurrentState(Game.Game game) {
             // Update The Seat List
             //await SendSeatsList(game);
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:402]: remove comment if not needed

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs [hunk 7]
```diff
@@ -474,5 +518,71 @@ public override async Task<StatusResponse> TakeSeat(TakeSeatRequest request, Ser
...
+            }
+
+            r.Success = true;
+            r.ErrorNum = (int)ErrorCode.SUCCESS;
+            r.ErrorText = "";
+
+            return r;
+        }
+
+        /*
```
Comment (deadwing88) [inline backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:566]: do we need these?

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 8]
```diff
@@ -110,13 +112,47 @@ export const GameProvider: React.FC = ({ children }) => {
                         }
                         state.eventEmitter.emit("seats", seatDetailsList);
                     }));
+                } else if (notification.hasStartGame()) {
+                    console.log('start game');
+                    // const obj: object = notification.toObject();
```
Comment (deadwing88) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:117]: maybe we should add these permanently under a more verbose log setting

File: backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs (hunk 7 above)
Comment (daveismith) [reply backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs:560]: Eventually, and this was me being lazy and bringing the function prototypes in now so I don't have to go looking for them later.

File: frontend/shoot-client/src/components/CreateGame/CreateGame.tsx [hunk 9]
```diff
@@ -17,6 +17,7 @@ const CreateGame: React.FC = () => {
 
     return (  
         <div>
+            <button onClick={ () => createGame(2) }>Create 2 Player Game</button><br />
```
[the remark this answers is not in the document]
Comment (timlaengert) [reply frontend/shoot-client/src/components/CreateGame/CreateGame.tsx:20]: Added to issues.

**Comment by daveismith** [comment]
Merging, outstanding issues will be cleaned up in future dev.

**Pull request merged**